### PR TITLE
Add more warning to the doc for resetOffsetAutomatically

### PR DESCRIPTION
**Pull request by jihoonson**
`resetOffsetAutomatically` is used only when it's allowed to ignore potential data loss or data duplicate. This PR adds more warning for this option. I additionally reformatted the table to look better in the markdown file. It should be same when it's rendered.

### CODE REVIEW

File: docs/content/development/extensions-core/kafka-ingestion.md [hunk 1]
```diff
@@ -130,31 +130,31 @@ A sample supervisor spec is shown below:
...
+| `resetOffsetAutomatically`        | Boolean        | This option _MUST_ not be set if you don't want to ignore missing data from Kafka. Once this is set, the supervisor _automatically_ resets the consumer offset for each Kafka partitions whenever the next offset it's trying to fetch is less than the earliest available offset in Kafka. The consumer offset will be reset to either the earliest or latest offset available in Kafka depending on `useEarliestOffset` property of `KafkaSupervisorIOConfig`. Please note that this could lead _data loss_ (when `useEarliestOffset` = false) or _data duplicate_ (when `useEarliestOffset` = true). The missing offsets in Kafka typically occurs when messages in Kafka are no longer available for consumption and therefore won’t be ingested into Druid. If set to false then ingestion for that particular partition will halt and manual intervention is required to correct the situation, please see [Reset Supervisor API](../../operations/api-reference.html#supervisors). | no (default == false) |
```
Comment (ccaominh) [inline docs/content/development/extensions-core/kafka-ingestion.md]: Typos:
- for each Kafka partitions -> for each Kafka partition
- depending on `useEarliestOffset` -> depending on the `useEarliestOffset`
- could lead _data loss_ -> could lead to _data loss_
- missing offsets in Kafka typically occurs -> missing offsets in Kafka typically occur

File: docs/content/development/extensions-core/kafka-ingestion.md [hunk 2]
```diff
@@ -130,31 +130,31 @@ A sample supervisor spec is shown below:
...
+| `indexSpecForIntermediatePersists`|                | defines segment storage format options to be used at indexing time for intermediate persisted temporary segments. this can be used to disable dimension/metric compression on intermediate segments to reduce memory required for final merging. however, disabling compression on intermediate segments might increase page cache use while they are used before getting merged into final segment published, see [IndexSpec](#indexspec) for possible values.                                                                                                                                                                                     | no (default = same as indexSpec)                                                                             |
```
Comment (ccaominh) [inline docs/content/development/extensions-core/kafka-ingestion.md]: Can capitalize the first letter of each sentence to be consistent with the other table rows.

File: docs/content/development/extensions-core/kinesis-ingestion.md [hunk 3]
```diff
@@ -126,37 +126,37 @@ A sample supervisor spec is shown below:
...
+| `indexSpecForIntermediatePersists`    |                | defines segment storage format options to be used at indexing time for intermediate persisted temporary segments. this can be used to disable dimension/metric compression on intermediate segments to reduce memory required for final merging. however, disabling compression on intermediate segments might increase page cache use while they are used before getting merged into final segment published, see [IndexSpec](#indexspec) for possible values.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           | no (default = same as indexSpec)                                                                             |
```
Comment (ccaominh) [inline docs/content/development/extensions-core/kinesis-ingestion.md]: Ditto for capitalization.

File: docs/content/development/extensions-core/kinesis-ingestion.md [hunk 4]
```diff
@@ -126,37 +126,37 @@ A sample supervisor spec is shown below:
...
+| `resetOffsetAutomatically`            | Boolean        | This option _MUST_ not be set if you don't want to ignore missing data from Kinesis. Once this is set, the supervisor _automatically_ resets the consumer offset for each Kinesis partitions whenever the next offset it's trying to fetch is less than the earliest available offset in Kinesis. The consumer offset will be reset to either the earliest or latest offset available in Kinesis depending on `useEarliestOffset` property of `KinesisSupervisorIOConfig`. Please note that this could lead _data loss_ (when `useEarliestOffset` = false) or _data duplicate_ (when `useEarliestOffset` = true). The missing offsets in Kinesis typically occurs when messages in Kinesis are no longer available for consumption and therefore won’t be ingested into Druid. If set to false then ingestion for that particular partition will halt and manual intervention is required to correct the situation, please see [Reset Supervisor API](../../operations/api-reference.html#supervisors). | no (default == false)                                                                                        |
```
Comment (ccaominh) [inline docs/content/development/extensions-core/kinesis-ingestion.md]: Typos:
- for each Kinesis partitions -> for each Kinesis partition
- depending on `useEarliestOffset` -> depending on the `useEarliestOffset`
- could lead _data loss_ -> could lead to _data loss_
- missing offsets in Kinesis typically occurs -> missing offsets in Kinesis typically occur

File: docs/content/development/extensions-core/kafka-ingestion.md (hunk 2 above)
Comment (jihoonson) [reply docs/content/development/extensions-core/kafka-ingestion.md]: Thanks, fixed.

File: docs/content/development/extensions-core/kinesis-ingestion.md (hunk 3 above)
Comment (jihoonson) [reply docs/content/development/extensions-core/kinesis-ingestion.md]: Thanks, fixed.

File: docs/content/development/extensions-core/kafka-ingestion.md (hunk 1 above)
Comment (jihoonson) [reply docs/content/development/extensions-core/kafka-ingestion.md]: Thank you for the review! The whole description is rewritten. Please take another look.

File: docs/content/development/extensions-core/kinesis-ingestion.md [hunk 5]
```diff
@@ -126,37 +126,37 @@ A sample supervisor spec is shown below:
...
+| `indexSpecForIntermediatePersists`    |                | Defines segment storage format options to be used at indexing time for intermediate persisted temporary segments. this can be used to disable dimension/metric compression on intermediate segments to reduce memory required for final merging. however, disabling compression on intermediate segments might increase page cache use while they are used before getting merged into final segment published, see [IndexSpec](#indexspec) for possible values.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                           | no (default = same as indexSpec)                                                                             |
```
Comment (ccaominh) [inline docs/content/development/extensions-core/kinesis-ingestion.md]: The first letter of the other sentences can also be capitalized ("This" and "However").

File: docs/content/development/extensions-core/kafka-ingestion.md [hunk 6]
```diff
@@ -130,31 +130,31 @@ A sample supervisor spec is shown below:
...
+| `indexSpecForIntermediatePersists`|                | Defines segment storage format options to be used at indexing time for intermediate persisted temporary segments. this can be used to disable dimension/metric compression on intermediate segments to reduce memory required for final merging. however, disabling compression on intermediate segments might increase page cache use while they are used before getting merged into final segment published, see [IndexSpec](#indexspec) for possible values.                                                                                                                                                                                     | no (default = same as indexSpec)                                                                             |
```
Comment (ccaominh) [inline docs/content/development/extensions-core/kafka-ingestion.md]: The first letter of the other sentences can also be capitalized ("This" and "However").

File: docs/content/development/extensions-core/kinesis-ingestion.md (hunk 5 above)
Comment (jihoonson) [reply docs/content/development/extensions-core/kinesis-ingestion.md]: Dang, thanks!

File: docs/content/development/extensions-core/kafka-ingestion.md (hunk 6 above)
Comment (jihoonson) [reply docs/content/development/extensions-core/kafka-ingestion.md]: Thanks, fixed.

**Comment by jihoonson** [comment]
This is a doc change and I believe CI failures are not legit. Will merge this shortly.

**Pull request merged**